### PR TITLE
Add logic to handle cancel_scope None

### PR DESCRIPTION
**Pull request by cicdw**
It seems that in rare circumstances we trigger an unhandled edge case (at least in tests, example [here](https://github.com/PrefectHQ/prefect/actions/runs/14145853194/job/39633057783?pr=17644)).

In an effort to hopefully make tests more resilient, this PR adds handling for this edge case.

### CODE REVIEW

**Review by zzstoatzz — APPROVED** [review]

File: src/prefect/_internal/concurrency/calls.py [hunk 1]
```diff
@@ -374,6 +374,11 @@ def _run_sync(self) -> Optional[Awaitable[T]]:
 
         except CancelledError:
             # Report cancellation
+            # in rare cases, enforce_sync_deadline raises CancelledError
+            # prior to yielding
+            if cancel_scope is None:
+                self.future.cancel()
+                return None
             if TYPE_CHECKING:
                 assert cancel_scope is not None
```
[the remark this answers is not in the document]
Comment (cicdw) [reply src/prefect/_internal/concurrency/calls.py:383]: hm i guess i don't know, i left it because `cancel_scope` should be not None thereafter, but if the type checker sees my `if` statement maybe it is? would you like me to remove it and see?

[the remark this answers is not in the document]
Comment (zzstoatzz) [reply src/prefect/_internal/concurrency/calls.py:383]: I would think your new condition makes the `assert` inside `if TYPE_CHECKING` redundant. after your new condition, the type checker should definitely know its not `None` and we don't need to trick it with `TYPE_CHECKING` anymore

[the remark this answers is not in the document]
Comment (cicdw) [reply src/prefect/_internal/concurrency/calls.py]: removed and CI is green ✅
